mm2src/coins/lp_coins.rs

```diff
@@ -1549,6 +1550,14 @@ pub trait ParseCoinAssocTypes {
 
     async fn my_addr(&self) -> Self::Address;
 
+    /// Converts coin `Self::Address` type into a properly formatted string representation.
+    ///
+    /// Don't use `to_string` directly on `Self::Address` types in generic TPU code!
+    /// It may produce abbreviated or non-standard formats (e.g. `ethereum_types::Address` will be like this `0x7cc9…3874`),
+    /// which are not guaranteed to be parsable back into the original `Address` type.
+    /// This function should ensure the resulting string is consistently formatted and fully reversible.
+    fn addr_to_string(&self, address: &Self::Address) -> String;
```

Comment:
nit: could this not be simply `fn my_addr_as_string(&self) -> String;`?
why convert arbitrary addresses?

Reply:
you are right, we can just use my_addr inside this function and do to string conversion
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/0f31d07d1eebf03a086c3cff657e524434581592